Source/WebCore/page/TextIndicator.cpp

```diff
@@ -67,6 +69,20 @@ Ref<TextIndicator> TextIndicator::create(const TextIndicatorData& data)
     return adoptRef(*new TextIndicator(data));
 }
 
+RefPtr<TextIndicator> TextIndicator::createWithNode(Node& node, OptionSet<TextIndicatorOption> options,  TextIndicatorPresentationTransition presentationTransition, FloatSize margin)
```

Comment:
Did you consider making this behavior a `TextIndicatorOption`, rather than a separate method?

Currently, it's not clear from this name of this method that it accounts for `user-select`, while `createWithRange` does not. There may also be other places where we do want to account for `user-select`, in which case it would be easier to add an option, than remember to use a specific method.

Reply:
New `TextIndicatorOption` (as opposed to create method) sounds good to me too!

Reply:
@pxlcoder  @whsieh how would this work exactly? Since `createWithNode` takes a `Node` as its parameter, whereas `createWithRange` takes a range

Reply:
I think the idea is that you would still pass in a range, but specify an option that allows `TextIndicator` to march up the DOM, to the closest `user-select: all` ancestor if needed.